default.nix

```diff
@@ -0,0 +1,251 @@
+{ pkgs ? import <nixpkgs> { }
+, stdenv ? pkgs.stdenv
+, lib ? pkgs.lib
+, ghostscript ? pkgs.ghostscript
+, ref ? null
+, doPdfGeneration ? true
+  # Attempts to render every page individually to make sure the dependencies
+  # are set correctly.
+, doCheck ? true
+  # Checks whether the generated output matches the checked-in SSS32.ps.
+  # When doing development you may want to shut this off to obtain the
+  # output file that you need to check in.
+  #
+  # Has no effect if doChecks is not set.
+, doOutputDiff ? true
+}:
+
+let
+  src =
+    if isNull ref
+    then lib.sourceFilesBySuffices ./. [ ".ps" ".inc" ]
+    else
+      fetchGit {
+        url = ./.;
+        inherit ref;
```

Comment:
After reviewing https://nix.dev/manual/nix/2.22/language/builtins#builtins-fetchGit I'm pretty sure you want `rev` instead of `ref`.

Reply:
Empirically `rev` seems to want a 40-character hash, while `ref` will accept short hashes, branch names, etc.